<?xml version="1.0"?>
<launch>
  <arg name="input/local/raw_control_cmd" />
  <arg name="input/local/shift_cmd" />
  <arg name="input/local/turn_signal_cmd" />

  <arg name="input/remote/raw_control_cmd" />
  <arg name="input/remote/shift_cmd" />
  <arg name="input/remote/turn_signal_cmd" />

  <arg name="output/current_selector_mode" default="~current_selector_mode"/>

  <arg name="output/raw_control_cmd" default="/external/external_cmd_selector/raw_control_cmd"/>
  <arg name="output/shift_cmd" default="/external/external_cmd_selector/shift_cmd"/>
  <arg name="output/turn_signal_cmd" default="/external/external_cmd_selector/turn_signal_cmd"/>

  <arg name="service/select_external_command" default="~select_external_command"/>

  <arg name="initial_selector_mode" default="0" doc="0: Local, 1: Remote" />

  <node pkg="external_cmd_selector" type="external_cmd_selector_node" name="external_cmd_selector" output="screen">
    <remap from="~input/local/raw_control_cmd" to="$(arg input/local/raw_control_cmd)" />
    <remap from="~input/local/shift_cmd" to="$(arg input/local/shift_cmd)" />
    <remap from="~input/local/turn_signal_cmd" to="$(arg input/local/turn_signal_cmd)" />

    <remap from="~input/remote/raw_control_cmd" to="$(arg input/remote/raw_control_cmd)" />
    <remap from="~input/remote/shift_cmd" to="$(arg input/remote/shift_cmd)" />
    <remap from="~input/remote/turn_signal_cmd" to="$(arg input/remote/turn_signal_cmd)" />

    <remap from="~output/current_selector_mode" to="$(arg output/current_selector_mode)" />

    <remap from="~output/raw_control_cmd" to="$(arg output/raw_control_cmd)" />
    <remap from="~output/shift_cmd" to="$(arg output/shift_cmd)" />
    <remap from="~output/turn_signal_cmd" to="$(arg output/turn_signal_cmd)" />

    <remap from="~service/select_external_command" to="$(arg service/select_external_command)" />

    <param name="initial_selector_mode" value="$(arg initial_selector_mode)" />
  </node>
</launch>
